src/Text/LLVM/AST.hs

```diff
@@ -1647,10 +1647,36 @@ data DISubprogram' lab = DISubprogram
 
 type DISubprogram = DISubprogram' BlockLabel
 
-data DISubrange = DISubrange
-  { disrCount      :: Int64
-  , disrLowerBound :: Int64
-  } deriving (Data, Eq, Generic, Ord, Show, Typeable)
+-- | The DISubrange is a Value subrange specification, usually associated with
+-- arrays or enumerations.
+--
+-- * Early LLVM: only 'disrCount' and 'disrLowerBound' were present, where both
+--   were a direct signed 64-bit value.  This corresponds to "format 0" in the
+--   bitcode encoding.
+--
+-- * LLVM 7: 'disrCount' changed to metadata representation ('ValMd').  The
+--   metadata representation should only be a signed 64-bit integer, a Variable,
+--   or an Expression.  This corresponds to "format 1" in the bitcode encoding.
+--
+-- * LLVM 11: 'disrLowerBound' was changed to a metadata representation and
+--   'disrUpperBound' and 'disrStride' were added (primarily driven by the
+--   addition of Fortran support in llvm).  All three should only be represented
+--   as a signed 64-bit integer, a Variable, or an Expression.  This corresponds
+--   to "format 2" in the bitcode encoding.  See
+--   https://github.com/llvm/llvm-project/commit/d20bf5a for this change.
+--
+-- Also see llvm-project/llvm/lib/Bitcode/Reader/MetadataLoader.cpp handling of
+-- bitc::METADATA_SUBRANGE (lines 1425-1461 in commit bbe8cd1) for how this is
```

Comment:
To save users from having to dig up this information themselves, we should just give a permalink here:
https://github.com/llvm/llvm-project/blob/bbe8cd13335300958b04db5318c31ff52714f96f/llvm/lib/Bitcode/Reader/MetadataLoader.cpp#L1435-L1461

Reply:
Done.